src/parquet/file/writer-internal.cc

```diff
@@ -77,13 +77,15 @@ int64_t SerializedPageWriter::WriteDataPage(int32_t num_rows, int32_t num_values
   int64_t compressed_size = uncompressed_size;
   std::shared_ptr<OwnedMutableBuffer> compressed_data = uncompressed_data;
   if (compressor_) {
-    // TODO(PARQUET-592): Add support for compression
-    // int64_t max_compressed_size = compressor_->MaxCompressedLen(
-    // uncompressed_data.size(), uncompressed_data.data());
-    // OwnedMutableBuffer compressed_data(compressor_->MaxCompressedLen(
-    // uncompressed_data.size(), uncompressed_data.data()));
+    uncompressed_ptr = uncompressed_data->mutable_data();
+    int64_t max_compressed_size = compressor_->MaxCompressedLen(
+        uncompressed_size, uncompressed_ptr);
+    compressed_data = std::make_shared<OwnedMutableBuffer>(max_compressed_size,
+        allocator_);
+    compressed_size = compressor_->Compress(uncompressed_size, uncompressed_ptr,
+        max_compressed_size, compressed_data->mutable_data());
```

Comment:
Can you use `uncompressed_data->data()` (since const is okay -- we are only mutating compressed_data)?